rundeckapp/grails-spa/packages/ui-trellis/src/app/components/readme-motd/EditProjectFile.spec.ts

```diff
@@ -0,0 +1,153 @@
+      authAdmin: true,
+      displayConfig: ["none"],
+      ...props,
+    },
+    global: {
+      mocks: {
+        $t: (msg) => msg,
+      },
+    },
+  });
+  await wrapper.vm.$nextTick();
+};
+
+describe("EditProjectFile", () => {
+  beforeEach(async () => {
+    await mountEditProjectFile();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it.each([
+    ["readme.md", "edit.readme.label"],
+    ["motd.md", "edit.motd.label"],
+  ])("renders the correct title for %s", async (filename, expectedTitle) => {
+    await mountEditProjectFile({ filename });
+    expect(wrapper.find('[data-test-id="title"]').text()).toContain(
+      expectedTitle,
+    );
+  });
+
+  it("renders file content when getFileText method returns successfully", async () => {
+    expect(wrapper.vm.fileText).toBe("sample file content");
+  });
+
+  it("handles failure when getFileText method fails", async () => {
+    (editProjectFileService.getFileText as jest.Mock).mockImplementationOnce(
+      () => Promise.reject(new Error("Failed to fetch file")),
+    );
+    wrapper.vm.notifyError = jest.fn();
+    await wrapper.vm.getFileText();
+    expect(wrapper.vm.notifyError).toHaveBeenCalledWith("Failed to fetch file");
+  });
+
+  it("handles failure when user edits the file and fails to save it", async () => {
+    (editProjectFileService.saveProjectFile as jest.Mock).mockRejectedValue(
+      new Error("Failed to save file"),
+    );
+    wrapper.vm.notifyError = jest.fn();
+    wrapper.vm.fileText = "new content";
+    await wrapper.find('[data-test-id="save"]').trigger("click");
+    expect(wrapper.vm.notifyError).toHaveBeenCalledWith("Failed to save file");
+  });
+
+  it("handles success when user edits the file and saves it", async () => {
+    wrapper.vm.fileText = "new content";
+    await wrapper.find('[data-test-id="save"]').trigger("click");
+    expect(
+      editProjectFileService.saveProjectFile as jest.Mock,
+    ).toHaveBeenCalledWith("default", "readme.md", "new content");
+  });
+
+  it("displays admin specific message and configuration link when user is an admin", async () => {
+    const footerText = wrapper.find(".card-footer").text();
+    expect(footerText).toContain("file.warning.not.displayed.admin.message");
+    expect(wrapper.find(".card-footer a").text()).toBe(
+      "project.configuration.label",
+    );
+  });
+
+  it("displays non-admin specific message when user is not an admin", async () => {
+    await mountEditProjectFile({
+      authAdmin: false,
+    });
+    expect(wrapper.text()).toContain(
+      "file.warning.not.displayed.nonadmin.message",
+    );
+  });
+  it("does not allow non-admin user to save the file", async () => {
+    await mountEditProjectFile({ authAdmin: false });
+    wrapper.vm.save = jest.fn();
+    wrapper.vm.getFileText = jest.fn();
+
+    wrapper.find('[data-test-id="save"]').trigger("click");
+    await wrapper.vm.$nextTick();
+    expect(wrapper.vm.save).not.toHaveBeenCalled();
+  });
```

Comment:
This test is wrong, there's no 'save' method, which is the reason why after the click, it wasn't called.

But checked now the component logic and the API, a non-admin user will be able to save the file, but they won't be able to adjust where the message will be displayed. So safe to delete this test.